File: 06_add_polynomials.c
#include <stdio.h>

int main() {
    int n;
    printf("Enter degree of polynomial: ");
    scanf("%d", &n);
    int poly1[n+1], poly2[n+1], sum[n+1];
    printf("Enter coefficients of 1st polynomial:\n");
    for(int i=0; i<=n; i++) scanf("%d", &poly1[i]);
    printf("Enter coefficients of 2nd polynomial:\n");
    for(int i=0; i<=n; i++) scanf("%d", &poly2[i]);
    for(int i=0; i<=n; i++) sum[i] = poly1[i] + poly2[i];
    printf("Sum polynomial: ");
    for(int i=n; i>=0; i--) {
        if(sum[i] != 0)
            printf("%d*x^%d ", sum[i], i);
    }
    return 0;
}

File: 06_add_polynomials.py
def addPolynomials(poly1, poly2):
    length = max(len(poly1), len(poly2))
    result = [0] * length
    for i in range(len(poly1)):
        result[i] += poly1[i]
    for i in range(len(poly2)):
        result[i] += poly2[i]
    return result

print("Polynomial Addition:", addPolynomials([5, 0, 10, 6], [1, 2, 4]))
